src/DataStructures/DataVector.hpp

```diff
@@ -54,15 +54,34 @@ using std::abs;  // NOLINT
 
 /*!
  * \ingroup DataStructuresGroup
- * \brief A class for storing data on a mesh.
+ * \brief Stores the nodal coefficients of a function on a Mesh.
  *
- * A Data holds an array of contiguous data and can be either owning (the array
- * is deleted when the Data goes out of scope) or non-owning, meaning it just
- * has a pointer to an array.
+ * \details Only use DataVector to represent function values on the
+ * computational grid. These nodal coefficients allow to approximate the
```

Comment:
DataVector can be used for any set of function values at distinct physical points (for example a set of points to which to interpolate) and not just those associated with a Mesh.  So we should not be talking about Lagrange polynomials, or that there must be an accompanying Mesh.